metricbeat/module/elasticsearch/elasticsearch.go

```diff
@@ -21,17 +21,54 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/url"
+	"sync"
+	"time"
 
+	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/metricbeat/helper"
 )
 
 // Global clusterIdCache. Assumption is that the same node id never can belong to a different cluster id
 var clusterIDCache = map[string]string{}
 
+// Global cache for license information. Assumption is that license information changes infrequently
+type _licenseCache struct {
+	sync.RWMutex
+	license  common.MapStr
+	cachedOn time.Time
+	ttl      time.Duration
+}
+
+var licenseCache = &_licenseCache{}
+
+func (c *_licenseCache) get() common.MapStr {
```

Comment:
nit: I would put the private and cache related code to the bottom of the file as I would assume most people come to the file to look at the public methods.